packages/core/src/integration-helpers.tsx

```diff
@@ -27,3 +32,111 @@ export const createIconComponentFromSimpleIconsSvgPath: (
     <path d={path} />
   </svg>
 )
+
+export type Status = 'idle' | 'loading' | 'ready' | 'error'
+export type ScriptElt = HTMLScriptElement | null
+
+export function useScript(
+  src: string,
+  attributes: { [key: string]: string }
+): Status {
+  const [status, setStatus] = useState<Status>(src ? 'loading' : 'idle')
+
+  useEffect(
+    () => {
+      if (!src) {
+        setStatus('idle')
+        return
+      }
+
+      // Fetch existing script element by src
+      // It may have been added by another instance of this hook
+      let script: ScriptElt = document.querySelector(`script[src="${src}"]`)
+
+      if (!script) {
+        // Create script
+        // eslint-disable-next-line consent-manager/do-not-inject-scripts
+        script = document.createElement('script')
+        script.src = src
+        script.async = true
+        script.defer = true
+        script.type = 'text/javascript'
+        script.setAttribute('data-status', 'loading')
+
+        if (attributes) {
+          Object.keys(attributes).forEach(name => {
+            // @todo why ts forces us to check here again?
+            script && script.setAttribute(name, attributes[name])
+          })
+        }
+
+        // Add script to document body
+        document.body.appendChild(script)
+
+        // Store status in attribute on script
+        // This can be read by other instances of this hook
+        const setAttributeFromEvent = (event: Event) => {
+          script?.setAttribute(
+            'data-status',
+            event.type === 'load' ? 'ready' : 'error'
+          )
+        }
+
+        script.addEventListener('load', setAttributeFromEvent)
+        script.addEventListener('error', setAttributeFromEvent)
+      } else {
+        // Grab existing script status from attribute and set to state.
+        setStatus(script.getAttribute('data-status') as Status)
+      }
+
+      // Script event handler to update status in state
+      // Note: Even if the script already exists we still need to add
+      // event handlers to update the state for *this* hook instance.
+      const setStateFromEvent = (event: Event) => {
+        setStatus(event.type === 'load' ? 'ready' : 'error')
+      }
+
+      // Add event listeners
+      script.addEventListener('load', setStateFromEvent)
+      script.addEventListener('error', setStateFromEvent)
+
+      // Remove event listeners on cleanup
+      return () => {
+        if (script) {
+          script.removeEventListener('load', setStateFromEvent)
+          script.removeEventListener('error', setStateFromEvent)
+          script.remove()
+        }
+      }
+    },
+    [src, attributes] // Only re-run effect if script src changes
+  )
+
+  return status
+}
+
+export function locateTracker(
```

Comment:
relevant, finds our tracker from the external script in the window object, to ensure user userTracker() hook returns the proper tracker